src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -0,0 +1,287 @@
+     * change per JVM session
+     *
+     * @param readDataHolderClass      class type of JNIReadDataHolderClass
+     * @param haplotypeDataHolderClass class type of JNIHaplotypeDataHolderClass
+     */
+    private native void jniInitializeClassFields(Class<?> readDataHolderClass, Class<?> haplotypeDataHolderClass);
+
+
+    public Boolean isAVXSupported() {
+        // use a grep command to check for AVX support
+        // grep exit code = 0 if a match was found
+        String command = runningOnMac ? "sysctl -a | grep machdep.cpu.features | grep -i avx" :
+                                        "grep -i avx /proc/cpuinfo";
+        try {
+            Process process = new ProcessBuilder("/bin/sh", "-c", command).start();
+            if (process.waitFor() != 0) {
+                return false;
+            }
+            if (process.exitValue() != 0) {
+                return false;
+            }
+        }
+        catch (InterruptedException | IOException e) {
+            return false;
+        }
+        return true;
+    }
+    
+    private static Boolean isVectorLoglessPairHMMLibraryLoaded = false;
+
+    //The constructor is called only once inside PairHMMLikelihoodCalculationEngine
+    public VectorLoglessPairHMM() throws UserException.HardwareFeatureException {
+        super();
+
+        synchronized (isVectorLoglessPairHMMLibraryLoaded) {
+            // if AVX is not supported, throw an exception
+            if (!isAVXSupported()) {
+                throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+            }
+
+            // Load the library and initialize the FieldIDs
+            if (!isVectorLoglessPairHMMLibraryLoaded) {
+                try
+                {
+                    //Try loading from Java's library path first
+                    //Useful if someone builds his/her own library and wants to override the bundled
+                    //implementation without modifying the Java code
+                    System.loadLibrary("VectorLoglessPairHMM");
+                    logger.info("libVectorLoglessPairHMM found in JVM library path");
+                } catch (UnsatisfiedLinkError ule) {
+                    //Could not load from Java's library path - try unpacking from jar
+                    try
+                    {
+                        logger.debug("libVectorLoglessPairHMM not found in JVM library path - trying to unpack from GATK jar file");
+                        String extension = runningOnMac ? ".dylib" : ".so";
+                        loadLibraryFromJar("/lib/libVectorLoglessPairHMM" + extension);
+                        logger.info("libVectorLoglessPairHMM unpacked successfully from GATK jar file");
+                    } catch (IOException ioe) {
+                        //Throw the UnsatisfiedLinkError to make it clear to the user what failed
+                        throw ule;
```

Comment:
maybe add an additional log message that it tried to to load the library from the jar and that also failed?